crates/router/src/connector/bluesnap.rs

```diff
@@ -0,0 +1,741 @@
+            headers::CONTENT_TYPE.to_string(),
+            self.get_content_type().to_string(),
+        ));
+        Ok(header)
+    }
+}
+
+impl ConnectorCommon for Bluesnap {
+    fn id(&self) -> &'static str {
+        "bluesnap"
+    }
+
+    fn common_get_content_type(&self) -> &'static str {
+        "application/json"
+    }
+
+    fn base_url<'a>(&self, connectors: &'a settings::Connectors) -> &'a str {
+        connectors.bluesnap.base_url.as_ref()
+    }
+
+    fn get_auth_header(
+        &self,
+        auth_type: &types::ConnectorAuthType,
+    ) -> CustomResult<Vec<(String, String)>, errors::ConnectorError> {
+        let auth: bluesnap::BluesnapAuthType = auth_type
+            .try_into()
+            .change_context(errors::ConnectorError::FailedToObtainAuthType)?;
+        let encoded_api_key =
+            consts::BASE64_ENGINE.encode(format!("{}:{}", auth.key1, auth.api_key));
+        Ok(vec![(
+            headers::AUTHORIZATION.to_string(),
+            format!("Basic {}", encoded_api_key),
+        )])
+    }
+
+    fn build_error_response(
+        &self,
+        res: Response,
+    ) -> CustomResult<ErrorResponse, errors::ConnectorError> {
+        logger::debug!(bluesnap_error_response=?res);
+        let response: bluesnap::BluesnapErrorResponse = res
+            .response
+            .parse_struct("BluesnapErrorResponse")
+            .change_context(errors::ConnectorError::ResponseDeserializationFailed)?;
+        Ok(ErrorResponse {
+            status_code: res.status_code,
+            code: match response.message.first() {
+                Some(first_error) => first_error.code.clone(),
+                _ => consts::NO_ERROR_CODE.to_string(),
+            },
+            message: match response.message.first() {
+                Some(first_error) => first_error.description.clone(),
+                _ => consts::NO_ERROR_MESSAGE.to_string(),
+            },
+            reason: None,
+        })
+    }
+}
+
+impl api::Payment for Bluesnap {}
+
+impl api::PreVerify for Bluesnap {}
+impl ConnectorIntegration<api::Verify, types::VerifyRequestData, types::PaymentsResponseData>
+    for Bluesnap
+{
+}
+
+impl api::PaymentVoid for Bluesnap {}
+
+impl ConnectorIntegration<api::Void, types::PaymentsCancelData, types::PaymentsResponseData>
+    for Bluesnap
+{
+    fn get_headers(
+        &self,
+        req: &types::PaymentsCancelRouterData,
+        connectors: &settings::Connectors,
+    ) -> CustomResult<Vec<(String, String)>, errors::ConnectorError> {
+        self.build_headers(req, connectors)
+    }
+
+    fn get_content_type(&self) -> &'static str {
+        self.common_get_content_type()
+    }
+
+    fn get_url(
+        &self,
+        _req: &types::PaymentsCancelRouterData,
+        connectors: &settings::Connectors,
+    ) -> CustomResult<String, errors::ConnectorError> {
+        Ok(format!(
+            "{}{}",
+            self.base_url(connectors),
+            "services/2/transactions"
```

Comment:
is this `2` api version? it's better to declare it at the top as const and use that everywhere

Reply:
The api_endpoints might change that's why complete url needs to be there.